Code cleanup, improved meta-tag handling and dependency updates

nakrovati opened the pull request:
1. Style: add `prettier` and format the code.
2. Refactor: handle static meta tags, icons and fonts in head.
3. Chore: update many dependencies: `vitepress`, `vue`, `x-satori`, `@vueuse/core`, `@octokit/rest`, etc.
4. Fix: fix a bug that caused the site to not build on `vitepress` v1.2.0+.
5. Refactor: replace `fast-globby` with `tinyglobby` - `tinyglobby` has fewer dependencies.
6. Chore: migrate `postcss` config to ESM.
7. Chore: remove `dotenv` from devDependencies - it is already built into `vitepress`.
8. Fix(docs): `markdownlint` errors.
9. Refactor: set moduleResolution to "bundler” in tsconfig.

nakrovati committed:
style: add prettier and format code

nakrovati committed:
chore: bump vitepress to 1.6.3

nakrovati committed:
fix: site build

When building the site there was an error `build error: Cannot read properties of undefined (reading 'replace')`.

nakrovati committed:
chore: replace fast-glob with tinyglobby

nakrovati committed:
chore: remove dotenv from devDependencies

Vitepress already includes dotenv.

nakrovati committed:
chore: bump deps

Update qrcode, x-satori, vue, vitepress-plugin-tabs, postcss-nesting, markdown-it, markdown-it-kdb, @types/node @types/markdown-it and @types/gtag.js.

nakrovati committed:
refactor: code cleanup

istudyatuni commented on .vitepress/config/en.ts:
```diff
@@ -1,7 +1,6 @@
 import type { DefaultTheme, LocaleConfig } from "vitepress";
 import type { CustomConfig, Theme } from "../theme/types";
 import { sections } from "../theme/plugins/section";
-import "dotenv/config";
```
this line imports env to `process.env`, see https://www.npmjs.com/package/dotenv

nakrovati replied:
I know, it's just that it's already built in https://vite.dev/guide/env-and-mode.html#env-files.

istudyatuni commented on .vitepress/config/hooks/generateMeta.ts:
```diff
@@ -9,29 +9,16 @@ function generateMeta(context: TransformContext, hostname: string) {
-  head.push(["link", { rel: "icon", href: "/logo-compact.svg?v=2", type: "image/svg+xml" }]);
-  head.push(["link", { rel: "apple-touch-icon", href: "/apple-touch-icon.png?v=2" }]);
-  head.push(["link", { rel: "manifest", href: "/site.webmanifest" }]);
-
```
I think it's better to return this to keep all such things in one place

nakrovati replied:
Vitepress recommends leaving static meta tags in head, because they work in development mode (changes to transformHead are only applied during site build, not development). https://vitepress.dev/reference/site-config#transformhead

istudyatuni commented on .vitepress/config/index.ts:
```diff
@@ -86,12 +86,20 @@ export default defineConfigWithTheme<Theme.Config>({
 
   vite: {
-    define: {
-      __VUE_PROD_HYDRATION_MISMATCH_DETAILS__: "false",
```
why removed?

nakrovati replied:
It's better to fix mismatch errors than to disable from checking in the compiler. Besides, I haven't seen a mismatch error appear anywhere.

nakrovati replied:
Oh, it's false by default ([docs](https://vuejs.org/api/compile-time-flags#VUE_PROD_HYDRATION_MISMATCH_DETAILS)), so it's not necessary.

istudyatuni commented on .vitepress/config/hooks/generateMeta.ts:
```diff
@@ -9,29 +9,16 @@ function generateMeta(context: TransformContext, hostname: string) {
-  if (pageData.frontmatter.type) {
-    head.push(["meta", { property: "og:type", content: pageData.frontmatter.type }]);
-  }
+  if (pageData.frontmatter.theme) head.push(["meta", { name: "theme-color", content: pageData.frontmatter.theme }]);
```
looks like this line wasn't formatted with prettier

nakrovati replied:
When I was fixing the tag generation, I thought that a single line is easier to read.

istudyatuni replied:
Hm, I think it's harder. At least it was for me when I first saw this

nakrovati replied:
Yeah, it's just a habit with Ruby. I can give it back if you want.

istudyatuni replied:
I think it will be better

Doesn't prettier format it?

nakrovati replied:
No, it doesn't format, I'll change it manually.

istudyatuni commented on .vitepress/theme/components/SearchBar.vue:
```diff
@@ -44,6 +39,7 @@ function onSearchBarClick(event: PointerEvent) {
     <form class="search-bar" @pointerup="onSearchBarClick($event)" @submit.prevent="">
       <label :title="placeholder" id="localsearch-label" :for="id"> <MagnifyingGlass width="18px" height="18[x" />q </label>
       <div v-if="backButton" class="search-actions before">
+        SOme
```
what is this?

nakrovati replied:
Accidentally typo, I'll remove it in a new commit

istudyatuni commented:
Thanks for the answers, sadly I can't "Resolve" them

nakrovati commented:
“resolve” as in approve PR or close conversations in comments? I have “resolve conversation” buttons.

nakrovati commented:
I can do that, should I close them all?

istudyatuni commented:
> should I close them all?

Yes, please

nakrovati committed:
refactor: code cleanup

refactor: set moduleResolution to 'bundler' in tsconfig

chore: remove @rollup/rollup-linux-x64-gnu from the dependencies list

fix: remove 'SOme' typo

nakrovati committed:
fix(docs): markdownlint errors

nakrovati committed:
chore: bump vueuse and octokit
